### PR TITLE
[AKS]: Add warnings to `az aks mesh` commands for out of support asm revision in use

### PR DESCRIPTION
**Pull request by deveshdama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks mesh`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by SanyaKochhar — APPROVED** [review]

**Comment by yonzhan** [comment]
@FumingZhang to help review

**Review by FumingZhang — APPROVED** [review]
LGTM

[live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=94234163&view=results) passed!

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -3305,6 +3316,28 @@ def _aks_mesh_update(
...
+            and service_mesh_profile.istio.revisions
+            and len(service_mesh_profile.istio.revisions) == 1
+        ):
+            revision = service_mesh_profile.istio.revisions[0]
+            supported_revisions = _aks_mesh_get_supported_revisions(cmd, client, mc.location)
+            if revision not in supported_revisions:
+                msg = (
+                    f"Istio mesh revision {revision} currently in use in your cluster is no longer supported.\n"
+                    "Please upgrade for continued support. Use `az aks mesh get-upgrades` to check for available "
+                    "upgrades.\n More information about mesh upgrades and version support can be found here:"
```
Comment (yanzhudd) [inline src/aks-preview/azext_aks_preview/custom.py:3336]:
```suggestion
                    "upgrades.\nMore information about mesh upgrades and version support can be found here:"
```

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**